### ISSUE TITLE
DOC: Add flycheck to complement flymake section?

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#6568]
In http://docs.astropy.org/en/latest/development/codeguide_emacs.html , there is mention of `flymake`. I propose also add a mention of `flycheck`, which is a successor but for newer Emacs (24+).

http://www.flycheck.org/en/latest/

**Comment by bsipocz** [comment]
Sounds good!

**Comment by mhvk** [comment]
From my unneeded issue #6681:
Currently, we recommend an emacs setup using flymake, pyflakes, and pep8 for syntax checking [1], but these are no longer developed (I noticed because I get warnings about syntax errors for `metaclass=...` arguments to class definitions). Instead, the recommendation is to use `flycheck` [2] with `flake8` (which we're also using on travis). We should update the documentation (and I need to update my setup...).

So, I'd say we should just change the recommendation. Also, @pllim, if you have already switched, what did you have to change? I'd love to save myself the time to find out what exactly is needed...

[1] http://docs.astropy.org/en/latest/development/codeguide_emacs.html
[2] http://www.flycheck.org/en/latest/

**Comment by mhvk** [comment]
p.s. Added the "Bug" label since the currently recommended setup gives an error for every file that includes a `metaclass` definition.

**Comment by pllim** [comment]
@mhvk , for `flycheck`, I basically followed instructions at http://www.flycheck.org/en/latest/user/installation.html#package-installation and then comment out those things again after it is installed (probably don't have to but I am paranoid about leaving it active). Once it is installed, I used the incantation at http://www.flycheck.org/en/latest/user/installation.html#use-package to turn it on.

p.s. The exact incantation I used is a bit convoluted because I still need `flymake` if older Emacs is detected.
